### PR TITLE
Implement the AnnexB Date part

### PR DESCRIPTION
**Pull request by kkristof**
JerryScript-DCO-1.0-Signed-off-by: Kristof Kosztyo kkosztyo.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.inc.h [hunk 1]
```diff
@@ -82,6 +82,9 @@ ROUTINE (LIT_MAGIC_STRING_SET_UTC_FULL_YEAR_UL, ecma_builtin_date_prototype_set_
 ROUTINE (LIT_MAGIC_STRING_TO_UTC_STRING_UL, ecma_builtin_date_prototype_to_utc_string, 0, 0)
 ROUTINE (LIT_MAGIC_STRING_TO_ISO_STRING_UL, ecma_builtin_date_prototype_to_iso_string, 0, 0)
 ROUTINE (LIT_MAGIC_STRING_TO_JSON_UL, ecma_builtin_date_prototype_to_json, 1, 1)
+ROUTINE (LIT_MAGIC_STRING_GET_YEAR_UL, ecma_builtin_date_prototype_get_year, 0, 0)
+ROUTINE (LIT_MAGIC_STRING_SET_YEAR_UL, ecma_builtin_date_prototype_set_year, 1, 1)
+ROUTINE (LIT_MAGIC_STRING_TO_GMT_STRING_UL, ecma_builtin_date_prototype_to_utc_string, 0, 0)
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.inc.h]: I think we need some guard for AnnexB. I don't think this legacy stuff is needed for compact profile.

**Comment by kkristof** [comment]
Hi, I have added a guard to the AnnexB and rebased the branch to the current master.

**Comment by zherczeg** [comment]
LGTM

**Comment by kkristof** [comment]
Hi, I have rebased the PR to the current master.

**Comment by egavrin** [comment]
Good to me

**Pull request merged**